components/file_upload/file_upload.jsx

```diff
@@ -635,8 +650,13 @@ export default class FileUpload extends PureComponent {
                 );
             });
             bodyAction = (
-                <React.Fragment>
+                <button
```

Comment:
Maybe I just don't understand what is going on here. But why do we have a button wrapping a file input field? Why does this have a separate onClick handler that checks if files can be uploaded?

Reply:
The button is wrapping the `input file` button.
The `input file` button is hidden as we do want it to be seen, thus it is not visible for users visually, but is still present.

We can't show a visual navigation on the `input file` button as that would mean making it visible, which we do not want to do, thus we are wrapping

The button allows us to navigate on it, we can make it a div as well with a `role="button"` but I think this is better.

Then when a user presses `enter` on the `button` (via keyboard navigation), we check if a user can upload a file, and then proceed to uploading it.

Reply:
However, I don't think we need to wrap the input in a button to do that, so I've separated the button out as a child, and also removed the extra function.